sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/ObjectHashAggregateExec.scala

```diff
@@ -67,7 +67,7 @@ case class ObjectHashAggregateExec(
     initialInputBufferOffset: Int,
     resultExpressions: Seq[NamedExpression],
```

Comment:
We need to add override modifier; otherwise the compiler will not capture it if we rename it in the future. 

Add `private override val` to all the four variables defined in BaseAggregateExec?

Reply:
I tried to rename one of the field `groupingExpressions` in sub-class `HashAggregationExec`, found the compiler and IDE will show error notification like:
```
class HashAggregationExec needs to be abstract, since value groupingExpressions in class BaseAggregateExec of type Seq[org.apache.spark.sql.catalyst.expressions.NamedExpression] is not defined
```

So maybe it's OK to keep it the concise style as I see in other places, e.g. `HashJoin` & `ShuffledHashJoinExec`?  
What do you think @gatorsmile @maropu

Reply:
Rather, I think it'd be better to add `override` in the other places, too.

Reply:
I see. Adding `override val` to the four common variables in 3a2f7dc116135198d3bd07aae03a941abb7703f0. `private` keyword will lead to same error as above. I can raise follow-up PR to refactor such base abstraction cases :- )

Reply:
Yea, the followup looks fine to me.